src/Analysis/Codelyzer.Analysis/CodeGraph.cs

```diff
@@ -147,223 +288,189 @@ private void AddEdges()
                             TargetNode = targetNode,
                             SourceNode = sourceNode
                         };
-                        sourceNode.Edges.Add(edge);
-                        targetNode.Edges.Add(edge);
+                        sourceNode.OutgoingEdges.Add(edge);
+                        targetNode.IncomingEdges.Add(edge);
                     }
                 });
-            });
-
-            // Add other types of edges
-            ustNodesByParent.ToList().ForEach(ustNodeKeyValue =>
+            }
+            else if (sourceNode.UstNode is InterfaceDeclaration interfaceDeclaration)
             {
-                var parentKey = ustNodeKeyValue.Key;
-                var sourceNode = GetClassNode(parentKey);
+                var baseTypeOriginalDefinition = interfaceDeclaration.BaseTypeOriginalDefinition;
```

Comment:
I think we should be looping through `interfaceDeclaration.BaseList` here instead of using `interfaceDeclaration.BaseTypeOriginalDefinition` since interfaces can still be subtypes of multiple other interfaces at once